BBallStatsTrack/src/com/example/bballstatstrack/models/utils/JSONDeserializer.java

```diff
@@ -0,0 +1,431 @@
+    {
+        ShotClass shotClass = ShotClass.valueOf( reader.nextString() );
+        ShotType shotType = ShotType.valueOf( reader.nextString() );
+        return new ShootEvent( shotClass, shotType, player, team );
+    }
+
+    private static SubstitutionEvent readEventSubstitution( JsonReader reader, Team team, Player player )
+            throws IOException
+    {
+        return new SubstitutionEvent( player, getPlayerFromTeam( team, reader.nextInt() ), team );
+    }
+
+    private static TurnoverEvent readEventTurnover( JsonReader reader, Team team, Player player ) throws IOException
+    {
+        return new TurnoverEvent( TurnoverType.valueOf( reader.nextString() ), player, team );
+    }
+
+    private static GameLog readGameLog( JsonReader reader ) throws IOException
+    {
+        GameLog gameLog = new GameLog();
+        reader.beginArray();
+        int index = 0;
+        while( reader.hasNext() )
+        {
+            gameLog.append( index, readPeriodLog( reader ) );
+            index++;
+        }
+        reader.endArray();
+        return gameLog;
+    }
+
+    private static List< Integer > readInGamePlayerNumbers( JsonReader reader ) throws IOException
+    {
+        List< Integer > list = new ArrayList< Integer >();
+        reader.beginArray();
+        while( reader.hasNext() )
+        {
+            list.add( reader.nextInt() );
+        }
+        reader.endArray();
+        return list;
+    }
+
+    private static List< GameEvent > readPeriodLog( JsonReader reader ) throws IOException
+    {
+        List< GameEvent > periodLog = new ArrayList< GameEvent >();
+        reader.beginArray();
+        while( reader.hasNext() )
+        {
+            reader.beginObject();
+            GameEvent event = readEvent( reader );
+            periodLog.add( event );
+            reader.endObject();
+        }
+        reader.endArray();
+        return periodLog;
+    }
+
+    private static Player readPlayer( JsonReader reader ) throws IOException
+    {
+        int number = Integer.MIN_VALUE;
+        String fullName = null;
+        int miss1pt = Integer.MIN_VALUE;
+        int miss2pt = Integer.MIN_VALUE;
+        int miss3pt = Integer.MIN_VALUE;
+        int made1pt = Integer.MIN_VALUE;
+        int made2pt = Integer.MIN_VALUE;
+        int made3pt = Integer.MIN_VALUE;
+        int offReb = Integer.MIN_VALUE;
+        int defReb = Integer.MIN_VALUE;
+        int assist = Integer.MIN_VALUE;
+        int to = Integer.MIN_VALUE;
+        int stl = Integer.MIN_VALUE;
+        int blk = Integer.MIN_VALUE;
+        int foul = Integer.MIN_VALUE;
+        int playingTimeSec = Integer.MIN_VALUE;
+        reader.beginObject();
+        String name;
+        while( reader.hasNext() )
+        {
+            name = reader.nextName();
+            if( name.equals( PlayerStats.NUMBER.toString() ) )
+            {
+                number = reader.nextInt();
```

Comment:
Player fields (besides the number and fullName) don't have setter methods. The only time they change value is via constructor, or via its methods that increment them by 1. It is designed this way to keep the fact that a player's stats doesn't change other than incrementing by 1 (example: gain 2 rebounds only by getting 1 rebound twice) I have updated the code the removes the need of the while loop and also the temporary initialization of the variables. They are now initialized to the correct value properly